### DISCUSSION TITLE
netclient - Fixed IP for Host

### DISCUSSION THREAD

**Discussion by Dyad-ITSS** [gravitl/netmaker#2641] [Q&A]
Hi All,

I've got new maker up and running and so far so good. Much quicker connection vs Tailscale which is what I needed.

I just have one problem, I have a host that I route to via a CaddyFile entry. server.mydomain.com -> 10.40.51.2. This works fine until the server restarts and the netclient reconnects. I then end up with another host with a new IP (10.40.51.3) which breaks the forward. 

How do I make sure the host gets or keeps the same IP every time it connects?

**Comment by mattkasun** [comment]
The host in question is running netclient?
Unless you leave and rejoin the network the host will have the same wireguard IP.

**Reply by Dyad-ITSS** [reply]
Correct, it's running netclient (docker).

If that machine has a power failure and restarts - it rejoins once docker boots up but then I have two instances of the machine (same host name) but new IP. It doesn't remove the older entry and take the same IP.

**Comment by Dyad-ITSS** [comment]
I think Egress is my answer - I've tried to set it up but I can't get it to work.

I'm not sure if I need to have a client in the mix but at the moment I have two hosts, host 1 is the VPS and host 2 is the local machine in my LAN. 

Does Egress work host to host?

**Reply by mattkasun** [reply]
yes you can communicate (via the egress gateway) to/from a netclient host to a machine on egress range.
